pp.c

```diff
@@ -7377,6 +7377,19 @@ PP(pp_floor)
     RETURN;
 }
 
+PP(pp_tainted)
+{
+    dSP;
+    dTARGET;
+    SV *arg = POPs;
+
+    SvGETMAGIC(arg);
+
+    sv_setbool_mg(TARG, SvTAINTED(arg));
+    PUSHs(TARG);
```

Comment:
There's no need to pop/push, since the top spot on the stack can simply be re-used, like in pp_not, and if we directly access PL_stack_sp like pp_not does we can skip the dSP and use NORMAL instead of RETURN.

Also like pp_not, there's no need to use TARG, we can simply use boolSV() to get &PL_sv_yes or &PL_sv_no.

Reply:
So I tried writing it like this:
```c
PP(pp_tainted)
{
    dSP;
    SV *arg = PL_stack_sp;

    SvGETMAGIC(arg);

    *PL_stack_sp = boolSV(SvTAINTED(arg));
    return NORMAL;
}
```

But I don't think the `SvGETMAGIC` is doing the right thing, it fails to call FETCH when I pass in a tied value ATM.

Reply:
```
SV *arg = PL_stack_sp;
```
should be:
```
SV *arg = *PL_stack_sp;
```

Reply:
Worked like a charm, thank you.